mautic/mautic
# [UX] Change contact details logic for not defined items

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#198 
| Issue(s) addressed                     | Fixes #... 



## Description

Dependency:
- #13780 
- #13909 

This PR ensures that users will have proper system feedback about features related to contact details.
Before, tabs were hidden and as a marketer, I often got confused with functionalities disappearing without knowing how to enable them.
Now, all features have a proper message when no data is available and Mautic feels more feature-rich.

![image](https://github.com/mautic/mautic/assets/116097999/ad06cf94-77d1-4f71-b3b9-2219cdce2537)

![image](https://github.com/mautic/mautic/assets/116097999/16de3a88-6082-4e23-a871-ced229ae0356)

I also added guidelines to the documentation, so devs can follow them when creating new features and also helping people to review PRs with clarity about how things need to be for approval.






---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open Contacts > New > Save & Close > Check all tabs in their details
